BigInteger/BigInteger.cs

```diff
@@ -1671,21 +1681,27 @@ public void genRandomBits(int bits, RNGCryptoServiceProvider rng)
         if (remBits != 0)
             dwords++;
 
-        if (dwords > maxLength)
-            throw (new ArithmeticException("Number of required bits > maxLength."));
+        if (dwords > maxLength || bits <= 0)
+            throw (new ArithmeticException("Number of required bits is not valid."));
 
-        byte[] randomBytes = new byte[4];
+        byte[] randomBytes = new byte[dwords * 4];
+        rng.GetBytes(randomBytes);
 
         for (int i = 0; i < dwords; i++)
-        {
-            rng.GetBytes(randomBytes);
-            data[i] = BitConverter.ToUInt32(randomBytes, 0);
-        }
+            data[i] = BitConverter.ToUInt32(randomBytes, i * 4);
+
+        for (int i = dwords; i < maxLength; i++)
+            data[i] = 0;
```

Comment:
Not necessary to zero the rest since `dataLength` tells us which byte is the last.